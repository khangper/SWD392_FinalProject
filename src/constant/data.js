export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
