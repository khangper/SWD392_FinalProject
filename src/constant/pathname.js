export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
