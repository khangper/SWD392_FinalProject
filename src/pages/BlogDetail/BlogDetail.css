

.blog-detail-icon {
  width: 10px;
  height: 10px;
  margin-top: 4.5px;
  margin-right: 10px;
}

.blog-detail-back-to-blog {
  position: relative;
  line-height: 20px;
  display: inline-block;
  min-width: 79px;
  white-space: nowrap;
}

.blog-detail-back-to-blog-link {
  align-self: stretch;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
}

.blog-detail-home {
  position: relative;
  line-height: 24px;
  color: #333;
  display: inline-block;
  min-width: 38px;
}

.blog-detail-breadcrumb-items,
.blog-detail-breadcrumb-items-1,
.blog-detail-single-blog-view {
  min-width: 6.1px;
  position: relative;
  line-height: 24px;
  display: inline-block;
}

.blog-detail-our-blog {
  position: relative;
  line-height: 24px;
  display: inline-block;
  color: #333;
  min-width: 54px;
  white-space: nowrap;
}

.blog-detail-single-blog-view {
  min-width: 104px;
  white-space: nowrap;
}

.blog-detail-breadcrumb-list {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 6.6px;
}

.blog-detail-heading-1 {
  position: relative;
  font-size: 24px;
  line-height: 30px;
  font-weight: 550;
  color: #333;
  white-space: nowrap;
}

.blog-detail-breadcrumb-content,
.blog-detail-breadcrumb-title {
  display: flex;

  align-items: flex-start;
}

.blog-detail-breadcrumb-title {
  padding: 0 20px 0 0;
  gap: 30px;
  color: rgba(33, 37, 41, 0.75);
  flex-direction: column;
  justify-content: flex-start;
}

.blog-detail-breadcrumb-content {
  width: 100%;
  max-width: 100%;
  flex-direction: row;
  justify-content: space-between;
  padding: 20px 430px 0 360px;
  height: 120px;
}

/*Content*/

.blog-detail-big-blog {
  height: 519.2px;
  flex: 1;
  position: relative;
  border-radius: 3px;
  max-width: 100%;
  object-fit: cover;
}

.blog-detail-blog-image {
  align-self: stretch;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 0 0 7px;
  box-sizing: border-box;
  max-width: 100%;
}

.blog-detail-views,
.blog-detail-point {
  position: relative;
  line-height: 20px;
  display: inline-block;
  min-width: 70px;
}

.blog-detail-point {
  min-width: 5.1px;
}

.blog-detail-view-count {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 3.5px;
}

.blog-detail-time-post {
  position: relative;
  line-height: 20px;
  display: inline-block;
  min-width: 97px;
}

.blog-detail-paragraph-text {
  align-self: stretch;
  height: 46px;
  border-bottom: 1px solid #e5e5e5;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-start;
  padding: 0 0 27px;
  gap: 7px;
}

.blog-detail-paragraph {
  margin: 0;
  font-size: 14px;
}

.blog-detail-content-container {
  position: relative;
  line-height: 26px;
}

.blog-detail-blog-meta {
  align-self: stretch;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 30px;
}

.blog-detail-heading-2 {
  position: relative;
  font-size: 17px;
  line-height: 26px;
  font-weight: 550;
  color: #333;
  display: inline-block;
  max-width: 100%;
}

.blog-detail-blog-content,
.blog-detail-blog-post {
  display: flex;
  align-items: flex-start;
  max-width: 100%;
}

.blog-detail-blog-post {
  flex-direction: column;
  justify-content: flex-start;
  gap: 23px;
  width: 780px;
  padding-right: 2.5px;
  padding-left: 2.5px;
}

.blog-detail-blog-content {
  align-self: stretch;
  flex-direction: row;
  justify-content: center;
  padding-top: 40px;
  box-sizing: border-box;
  background-color: #f7f7f7;
}

.blog-detail-header-parent {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  max-width: 100%;
  flex-shrink: 0;
  text-align: left;
  font-size: 14px;
  color: #686f7a;
  flex: 1;
  flex-direction: column;
  box-sizing: border-box;
}

.blog-detail-previous,
.blog-detail-next {
  display: inline-block;
  line-height: 20px;
  font-weight: 550;
  margin: 0 !important;
  color: #333;
}

.blog-detail-footer-content-1,
.blog-detail-footer-content-2 {
  display: flex;
  flex-direction: column;
  width: 200px;
  gap: 10px;
}

.blog-detail-footer-content-2 {
  text-align: right;
}

.blog-detail-icon-prev,
.blog-detail-icon-next {
  width: 17px;
  height: 24px;
  color: #333;
  margin-top: 25px;
}

.blog-detail-content-1,
.blog-detail-content-2 {
  line-height: 20px;
}

.blog-detail-footer-container-1,
.blog-detail-footer-container-2 {
  display: flex;
  flex-direction: row;
  padding: 60px 0 30px;
  cursor: pointer;
  gap: 20px;
}

.blog-detail-footer-container-1:hover, .blog-detail-footer-container-2:hover{
  gap: 10px;
}

.blog-detail-footer-parent {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  height: 180px;
  max-width: 100%;
  background-color: #f7f7f7;
  padding: 0 360px;
  box-sizing: border-box;
}

@media screen and (max-width: 1350px) {
  .blog-detail-breadcrumb-content,
  .blog-detail-footer-parent {
    max-width: 100%;
    box-sizing: border-box;
    padding-right: 180px;
    padding-left: 180px;
  }
}

@media screen and (max-width: 900px) {
  .blog-detail-breadcrumb-content {
    max-width: 100%;
    box-sizing: border-box;
    padding-left: 80px;
    padding-right: 80px;
  }

  .blog-detail-footer-parent {
    max-width: 100%;
    box-sizing: border-box;
    padding-left: 40px;
    padding-right: 40px;
  }
}
