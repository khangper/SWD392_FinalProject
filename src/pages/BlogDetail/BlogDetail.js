import React, { useEffect } from "react";
import "./BlogDetail.css";
import Header from "../../components/Header/Header";
import { MdKeyboardDoubleArrowLeft } from "react-icons/md";
import big_blog from "../../assets/big_blog.jpg";
import { MdKeyboardDoubleArrowRight } from "react-icons/md";
import Footer from "../../components/Footer/Footer";
import { PATH_NAME } from "../../constant/pathname";
import { Link, useParams } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { fetchBlogDetailRequest } from "../../redux/reduxActions/blogActions/BlogDetailAction";
import numeral from "numeral";

const BlogDetail = () => {
  const dispatch = useDispatch();
  const { id } = useParams();
  const { blog } = useSelector((state) => state.blogDetail);

  useEffect(() => {
    dispatch(fetchBlogDetailRequest(id));
  }, [dispatch, id]);

  if (!blog) {
    return null;
  }

  return (
    <div className="blog-detail">
      <Header />
      <main className="blog-detail-parent">
        <div className="blog-detail-header-parent">
          <div className="blog-detail-breadcrumb-content">
            <div className="blog-detail-breadcrumb-title">
              <div className="blog-detail-breadcrumb-list">
                <Link to={PATH_NAME.HOME} className="blog-detail-home">
                  Home
                </Link>
                <div className="blog-detail-breadcrumb-items">/</div>
                <Link to={PATH_NAME.OUR_BLOG} className="blog-detail-our-blog">
                  Our Blog
                </Link>
                <div className="blog-detail-breadcrumb-items-1">/</div>
                <div className="blog-detail-single-blog-view">
                  Single Blog View
                </div>
              </div>
              <h3 className="blog-detail-heading-1">
                {blog.blog_detail.title}
              </h3>
            </div>
            <div className="blog-detail-back-to-blog-link">
              <MdKeyboardDoubleArrowLeft className="blog-detail-icon" />
              <Link to="/our-blog" className="blog-detail-back-to-blog">
                Back to Blog
              </Link>
            </div>
          </div>

          <div className="blog-detail-blog-content">
            <div className="blog-detail-blog-post">
              <div className="blog-detail-blog-image">
                <img
                  src={blog.blog_detail.thumbnails}
                  alt=""
                  className="blog-detail-big-blog"
                />
              </div>
              <div className="blog-detail-blog-meta">
                <div className="blog-detail-paragraph-text">
                  <div className="blog-detail-view-count">
                    <div className="blog-detail-views">
                      {numeral(blog.blog_detail.viewCount)
                        .format("0.a")
                        .replace(/m/g, "M")
                        .replace(/k/g, "K")}{" "}
                      views
                    </div>
                    <div className="blog-detail-point">â€¢</div>
                  </div>
                  <div className="blog-detail-time-post">
                    {blog.blog_detail.createdAt}
                  </div>
                </div>
                <div className="blog-detail-content-container">
                  <p className="blog-detail-paragraph">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Curabitur volutpat maximus pellentesque. Integer sem enim,
                    luctus at nibh at, condimentum sagittis sapien. Sed tempus
                    ipsum erat, sit amet efficitur velit interdum eu. Vestibulum
                    hendrerit id dolor eu scelerisque. Phasellus ex dui,
                    consequat nec feugiat eu, dapibus eget ante. Sed sodales
                    interdum dui, at euismod mi feugiat hendrerit. Suspendisse
                    auctor libero in tempor mollis. Nulla et dolor velit.
                    Aliquam sit amet luctus quam.
                  </p>
                </div>
                <div className="blog-detail-heading-2">
                  Why did you decide to teach on Haka?
                </div>
                <div className="blog-detail-content-container">
                  <p className="blog-detail-paragraph">
                    Nam a egestas libero, eget eleifend turpis. Sed id ipsum a
                    ipsum aliquam laoreet sit amet sit amet nibh. Proin dapibus,
                    libero sed posuere rhoncus, orci mi haka enim, at accumsan
                    eros massa lacinia mi. Nunc eget finibus felis, volutpat
                    malesuada sem. Aliquam ac nisl pellentesque, varius neque
                    sit amet, porttitor nunc. Nullam elit tellus, dapibus non
                    eleifend sed, hendrerit eget velit. Aliquam ut felis dictum,
                    tincidunt magna vitae, aliquam massa. In porttitor tristique
                    quam, non dignissim sapien pharetra ultrices. Cras non ante
                    non velit mollis mollis. Pellentesque habitant morbi
                    tristique senectus et netus et malesuada fames ac turpis
                    egestas. Quisque et bibendum urna, eget consequat sapien.
                    Integer sed condimentum nibh. Integer id neque tristique,
                    lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor
                    ac rutrum eget, volutpat a velit. Curabitur et enim quis
                    diam congue dictum et vitae dui. Nulla tortor orci, luctus a
                    pretium vel, ultrices porta nisl.
                  </p>
                </div>
                <div className="blog-detail-content-container">
                  <p className="blog-detail-paragraph">
                    Etiam lobortis dictum tincidunt. Interdum et malesuada fames
                    ac ante ipsum primis in faucibus. Suspendisse ultricies
                    efficitur dui, suscipit tempus elit condimentum quis. Duis
                    sed vestibulum tortor, eget haka odio.
                  </p>
                </div>

                <div className="blog-detail-heading-2">
                  Did you have any prior on- or offline teaching experience
                  prior to publishing your first Haka course?
                </div>
                <div className="blog-detail-content-container">
                  <p className="blog-detail-paragraph">
                    Morbi lectus nunc, lacinia ut consequat a, semper vel erat.
                    Duis ut lacus nec dui sodales mattis. Mauris tellus dolor,
                    pulvinar sit amet pretium a, malesuada sed tellus. Aliquam
                    ultrices elit neque, quis lacinia ex porttitor non. Donec ac
                    iaculis turpis. Nulla lacinia enim quis orci aliquam, non
                    haka urna pellentesque. Class aptent taciti sociosqu ad
                    litora torquent per conubia nostra, per inceptos himenaeos.
                    Phasellus in mi a nisi auctor interdum. Vivamus faucibus
                    magna sed elit interdum consequat. Vestibulum eu tortor vel
                    ante feugiat faucibus quis et urna. Quisque interdum ac enim
                    eu tempus. Fusce viverra, lectus egestas tincidunt haka,
                    tortor sapien convallis metus, vitae auctor risus metus vel
                    nisi. Aenean dapibus bibendum mauris ut iaculis.
                  </p>
                </div>
                <div className="blog-detail-heading-2">
                  What are the most rewarding aspects of teaching on Haka?
                </div>

                <div className="blog-detail-content-container">
                  <p className="blog-detail-paragraph">
                    Quisque et bibendum urna, eget consequat sapien. Integer sed
                    condimentum nibh. Integer id neque tristique, lobortis massa
                    ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum
                    eget, volutpat a velit. Curabitur et enim quis diam congue
                    dictum et vitae dui. Nulla tortor orci, luctus a pretium
                    vel, ultrices porta nisl.
                  </p>
                </div>
              </div>
            </div>
          </div>

          <div className="blog-detail-footer-parent">
            <div className="blog-detail-footer-container-1">
              <MdKeyboardDoubleArrowLeft className="blog-detail-icon-prev" />
              <div className="blog-detail-footer-content-1">
                <h3 className="blog-detail-previous">Previous</h3>
                <p className="blog-detail-content-1">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                </p>
              </div>
            </div>

            <div className="blog-detail-footer-container-2">
              <div className="blog-detail-footer-content-2">
                <h3 className="blog-detail-next">Next</h3>
                <p className="blog-detail-content-2">
                  Vestibulum vulputate nulla quis dignissim ultricies.
                </p>
              </div>
              <MdKeyboardDoubleArrowRight className="blog-detail-icon-next" />
            </div>
          </div>
        </div>
      </main>
      <Footer />
    </div>
  );
};

export default BlogDetail;
