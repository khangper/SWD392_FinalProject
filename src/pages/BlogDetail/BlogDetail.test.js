import React from "react";
import { render, screen } from "@testing-library/react";
import BlogDetail from "../BlogDetail/BlogDetail";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("BlogDetail", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the BlogDetail component", async () => {
    const { container } = renderWithProviders(<BlogDetail />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
