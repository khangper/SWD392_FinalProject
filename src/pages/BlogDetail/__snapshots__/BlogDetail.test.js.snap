// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogDetail renders the BlogDetail component 1`] = `<div />`;
